notion-server/src/ethereum/mod.rs

```diff
@@ -663,4 +724,72 @@ mod tests {
             "value saved should match initial args passed in"
         );
     }
+
+    fn produce_block(
+        test_name: &str,
+        wasm_code: &[u8],
+        initial_state: &str,
+        data: &str,
+        expected_end_state: &str,
+    ) {
+        let mut simulation = Simulation::new();
+
+        assert_eq!(
+            0,
+            simulation
+                .create_execution_environment(args::CreateExecutionEnvironment {
+                    execution_environment: args::ExecutionEnvironment {
+                        base64_encoded_wasm_code: base64::encode(wasm_code),
+                        base64_encoded_initial_state: base64::encode(
+                            &initial_state.from_hex().unwrap()
+                        ),
+                    }
+                })
+                .unwrap()
+        );
+
+        assert_eq!(0, simulation.create_shard_chain(args::CreateShardChain {}));
+
+        assert_eq!(
+            0,
+            simulation
+                .create_shard_block(args::CreateShardBlock {
+                    shard_chain_index: 0,
+                    shard_block: args::ShardBlock {
+                        transactions: vec![args::ShardTransaction {
+                            base64_encoded_data: base64::encode(&data.from_hex().unwrap()),
+                            ee_index: 0,
+                        }]
+                    }
+                })
```

Comment:
So there's nothing technically wrong with this, but I think I'd recommend using some temporary variables to reduce indentation and increase readability.

Reply:
done.